src/locator/al-route.types.ts

```diff
@@ -413,44 +438,46 @@ export class AlRoute {
+     *      a) nested conditions
+     *      b) route parameters or parameter expressions
+     *      c) path matches
+     *      d) entitlements or other externally calculated conditions
+     *  And then uses a simple reducer to roll up all results based on the condition rule, which defaults to 'all'.
      */
     evaluateCondition( condition:AlRouteCondition|boolean ):boolean {
         if ( typeof( condition ) === 'boolean' ) {
             return condition;
         }
+
+        let evaluations:boolean[] = [];
         if ( condition.rule && condition.conditions ) {
-            //  This condition is a group of other conditions -- evaluate it internally
-            let total = 0;
-            let passed = 0;
-            condition.conditions.forEach( child => {
-                total++;
-                passed += this.evaluateCondition( child ) ? 1 : 0;
-            } );
-            if ( condition.rule === "any" ) {
-                return (passed > 0);
-            } else if ( condition.rule === "all" ) {
-                return (passed === total);
-            }
-            return (passed === 0);
+            evaluations = evaluations.concat( condition.conditions.map( condition => this.evaluateCondition( condition ) ) );
         }
 
-        let truthful = true;
         if ( condition.parameters ) {
-            //  Evaluates true only if all of the referenced route parameters exist
-            truthful = truthful && condition.parameters.reduce<boolean>( ( evaluation, parameterExpression ):boolean => {
-                    return evaluation && this.evaluateParameterExpression( parameterExpression );
-                }, true );
+            evaluations = evaluations.concat( condition.parameters.map( parameterExpression => this.evaluateParameterExpression( parameterExpression ) ) );
         }
         if ( condition.path_matches ) {
             //  Evaluates true only if the current path matches a given regular expression
-            truthful = truthful && this.evaluatePathMatch( condition.path_matches );
+            evaluations.push( this.evaluatePathMatch( condition.path_matches ) );
         }
         if ( condition.entitlements ) {
             //  This condition refers to entitlement or other externally managed data -- ask the host to evaluate it.
-            truthful = truthful && this.host.evaluate( condition );
+            evaluations.push( this.host.evaluate( condition ) );
+        }
+        if ( condition.environments ) {
+            evaluations.push( condition.environments.includes( AlLocatorService.getContext().environment ) );
+        }
+        if ( condition.rule === 'none' ) {
+            return ! evaluations.reduce( ( alpha, value ) => alpha || value, false );
+        } else if ( condition.rule === 'any' ) {
+            return evaluations.reduce( ( alpha, value ) => alpha || value, false );
+        } else {
+            return evaluations.reduce( ( alpha, value ) => alpha && value, true );
```

Comment:
these can be changed to .some and .every to provide short circuiting

Reply:
Nice, done